omim2obo/main.py

```diff
@@ -231,18 +235,55 @@ def run(use_cache: bool = False):
             graph.add((OMIM[mim_number], RDFS.subClassOf, OMIMPS[ps_id]))
 
     # Morbid map (cyto locations)
-    morbid_map: Dict[str, List[str]] = parse_morbid_map(
-        retrieve_mim_file('morbidmap.txt', download_files_tf))
-    for mim_number in morbid_map:
-        phenotype_mim_number, cyto_location = morbid_map[mim_number]
-        if phenotype_mim_number:
-            # RO:0003303 == 'causes condition' https://www.ebi.ac.uk/ols/ontologies/ro/properties?iri=http://purl.obolibrary.org/obo/RO_0003303
-            graph.add((OMIM[mim_number], RO['0003303'], OMIM[phenotype_mim_number]))
+    morbid_map, morbid_map_issues = parse_morbid_map(get_mim_file('morbidmap.txt', download_files_tf))
+    if morbid_map_issues:
+        issues['morbid_map'] = {'issue:multipleValuesDetected': morbid_map_issues}
+    for mim_number, mim_data in morbid_map.items():
+        # todo?: unused `mim_data` keys:
+        #   'gene_mim_number': Should this be used somewhere?
+        #   'gene_symbols':  Should this be used somewhere?
+        #   'phenotype_label': I think this already added to graph because the MIM# and label shared in other file(s)
+        cyto_location: str = mim_data['cyto_location']
         if cyto_location:
             # What's 9606chr - joeflack4 2021/11/11
             chr_id = '9606chr' + cyto_location
-            # What's RO['0002525'] - joeflack4 2021/11/11
+            # RO:0002525 (is subsequence of)
+            # https://www.ebi.ac.uk/ols/ontologies/ro/properties?iri=http://purl.obolibrary.org/obo/RO_0002525
             graph.add((OMIM[mim_number], RO['0002525'], CHR[chr_id]))
+        assocs: Dict = mim_data['phenotype_associations']
+        if len(assocs) == 1:
+            # RO:0003303 (causes condition)
+            # https://www.ebi.ac.uk/ols/ontologies/ro/properties?iri=http://purl.obolibrary.org/obo/RO_0003303
+            # TODO: what to do? This should always be a MIM number, I would think. But there is not always a MIM number
+            assoc = list(assocs.values())[0]
+            pheno_id = assoc['phenotype_identifier']
+            try:
+                pheno_id = int(pheno_id)
+                graph.add((OMIM[mim_number], RO['0003303'], OMIM[pheno_id]))
+            except ValueError:
+                # Report issue (No MIM # In Phenotype label), e.g.
+                #   Phenotype	Gene Symbols	MIM Number	Cyto Location
+                #   3p- syndrome (4)	DEL3pterp25, C3DELpterp25	613792	3pter-p25
+                if 'morbid_map' not in issues:
```

Comment:
This is another case of ugly code where I'm checking for issues and reporting them. This will change once we've figured out what we want to do.

Reply:
Done with reporting on these issues, removing this code.